src/backend/env/proposals.rs

```diff
@@ -283,23 +289,31 @@ impl Payload {
     }
 }
 
-pub fn propose(
+pub fn create_proposal(
     state: &mut State,
     caller: Principal,
-    description: String,
+    post_id: PostId,
     mut payload: Payload,
     time: u64,
 ) -> Result<u32, String> {
+    if Post::get(state, &post_id).is_none() {
+        return Err("post not found".to_string());
+    }
     payload.validate(state)?;
+
```

Comment:
Would it be better to put line 314 `!user.stalwart {` before heavy operations (like hashing the binary) ?

```suggestion
    if !user.stalwart {
        return Err("only stalwarts can create proposals".to_string());
    }
```